src/standard.ts

```diff
@@ -1,202 +1,196 @@
-import { equalsBytes, hexToBytes } from 'ethereum-cryptography/utils';
-import { Bytes, compareBytes, hex } from './bytes';
-import { getProof, isValidMerkleTree, makeMerkleTree, processProof, renderMerkleTree, MultiProof, getMultiProof, processMultiProof } from './core';
-import { MerkleTreeOptions, defaultOptions } from './options';
-import { checkBounds } from './utils/check-bounds';
-import { throwError } from './utils/throw-error';
-import { standardLeafHash } from './utils/standard-leaf-hash';
-
-interface StandardMerkleTreeData<T extends any[]> {
-  format: 'standard-v1';
-  tree: string[];
-  values: {
-    value: T;
-    treeIndex: number;
-  }[];
-  leafEncoding: string[];
-}
+import { BytesLike, HexString, toHex, isBytes, isBytesLike } from "./bytes";
 
-export class StandardMerkleTree<T extends any[]> {
-  private readonly hashLookup: { [hash: string]: number };
+import { MultiProof, processProof, processMultiProof } from "./core";
 
-  private constructor(
-    private readonly tree: Bytes[],
-    private readonly values: { value: T, treeIndex: number }[],
-    private readonly leafEncoding: string[],
-  ) {
-    this.hashLookup =
-      Object.fromEntries(values.map(({ value }, valueIndex) => [
-        hex(standardLeafHash(value, leafEncoding)),
-        valueIndex,
-      ]));
-  }
+import {
+  SimpleMerkleTree,
+  SimpleMerkleTreeData,
+  SimpleMerkleTreeValue,
+} from "./simple";
 
-  static of<T extends any[]>(values: T[], leafEncoding: string[], options: MerkleTreeOptions = {}) {
-    const sortLeaves = options.sortLeaves ?? defaultOptions.sortLeaves;
+import { defaultAbiCoder } from "@ethersproject/abi";
+import { keccak256 } from "@ethersproject/keccak256";
 
-    const hashedValues = values.map((value, valueIndex) => ({ value, valueIndex, hash: standardLeafHash(value, leafEncoding) }));
+import { StandardMerkleTreeOptions } from "./options";
+import { throwError } from "./utils/throw-error";
 
-    if (sortLeaves) {
-      hashedValues.sort((a, b) => compareBytes(a.hash, b.hash));
-    }
+export type StandardMerkleTreeValue = SimpleMerkleTreeValue;
 
-    const tree = makeMerkleTree(hashedValues.map(v => v.hash));
+export type StandardMerkleTreeData = SimpleMerkleTreeData & {
```

Comment:
`SimpleMerkleTreeData` is 
```typescript
export type SimpleMerkleTreeValue = {
  value: BytesLike[];
  treeIndex: number;
};
export type SimpleMerkleTreeData = {
  format: string;
  tree: HexString[];
  values: SimpleMerkleTreeValue[];
};
```
which means that the proposed `StandardMerkleTreeData` uses the values to be `HexString[]`.

Note that the current type is:
```typescript
interface StandardMerkleTreeData<T extends any[]> {
  format: 'standard-v1';
  tree: string[];
  values: {
    value: T;
    treeIndex: number;
  }[];
  leafEncoding: string[];
}
``` 
Where the values are `any[]`.

This means that the proposed change is breaking backward compatibility, particularly is loses the ability to load old dump. That is a big no to me.

Reply:
> This means that the proposed change is breaking backward compatibility, particularly is loses the ability to load old dump. 

I was completely unfamiliar with this. It's also a no for me. I'll mark it as draft before reviewing your iteration and updating this one with `any` type.